modules/schema/src/main/java/org/apache/ignite/internal/schema/Row.java

```diff
@@ -17,69 +17,55 @@
 
 package org.apache.ignite.internal.schema;
 
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
 import java.util.BitSet;
 import java.util.UUID;
 
 /**
+ * Schema-aware row.
+ *
  * The class contains non-generic methods to read boxed and unboxed primitives based on the schema column types.
  * Any type conversions and coercions should be implemented outside the row by the key-value or query runtime.
  * When a non-boxed primitive is read from a null column value, it is converted to the primitive type default value.
  */
-public abstract class Row {
-    /** */
-    public static final int SCHEMA_VERSION_OFFSET = 0;
-
-    /** */
-    public static final int FLAGS_FIELD_OFFSET = SCHEMA_VERSION_OFFSET + 2;
-
-    /** */
-    public static final int KEY_HASH_FIELD_OFFSET = FLAGS_FIELD_OFFSET + 2;
-
-    /** */
-    public static final int KEY_CHUNK_OFFSET = KEY_HASH_FIELD_OFFSET + 4;
-
-    /** */
-    public static final int TOTAL_LEN_FIELD_SIZE = 4;
-
-    /** */
-    public static final int VARLEN_TABLE_SIZE_FIELD_SIZE = 2;
-
-    /** */
-    public static final int VARLEN_COLUMN_OFFSET_FIELD_SIZE = 2;
+public class Row implements BinaryRow {
+    /** Schema descriptor. */
+    private final SchemaDescriptor schema;
 
-    /** */
-    public static final class RowFlags {
-        /** Tombstone flag. */
-        public static final int TOMBSTONE = 1;
+    /** Binary row. */
+    private final BinaryRow row;
 
-        /** Null-value flag. */
-        public static final int NULL_VALUE = 1 << 1;
+    /**
+     * Constructor.
+     *
+     * @param schema Schema.
+     * @param row Binary row representation.
+     */
+    public Row(SchemaDescriptor schema, BinaryRow row) {
+        assert row.schemaVersion() == schema.version();
 
-        /** Stub. */
-        private RowFlags() {
-        }
+        this.row = row;
+        this.schema = schema;
     }
 
-    /** Schema descriptor for which this row was created. */
-    private final SchemaDescriptor schema;
-
     /**
-     * @param schema Schema instance.
+     * @return Row schema.
      */
-    protected Row(SchemaDescriptor schema) {
-        this.schema = schema;
+    public SchemaDescriptor rowSchema() {
+        return schema;
     }
 
     /**
      * @return {@code True} if row has non-null value, {@code false} otherwise.
      */
-    public boolean hasValue() {
-        short flags = readShort(FLAGS_FIELD_OFFSET);
-
-        return (flags & (RowFlags.NULL_VALUE | RowFlags.TOMBSTONE)) == 0;
+    @Override public boolean hasValue() {
+        return row.hasValue();
     }
 
     /**
+     *
```

Comment:
Please describe the contract of this method.
What happens if the required column has a different type.
What happens if the row does not have a value.